sunpy/net/base_client.py

```diff
@@ -109,65 +112,185 @@ def show(self, *cols):
+    hide_keys = TableAttribute()
+
+    # This is a work around for https://github.com/astropy/astropy/pull/11217
+    # Remove when min astropy version is > 4.2.1
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        for attr in list(kwargs):
+            descr = getattr(self.__class__, attr, None)
+            if isinstance(descr, TableAttribute):
+                setattr(self, attr, kwargs.pop(attr))
 
+    @deprecated("2.1", "The object is a table.")
     def build_table(self):
-        return self.table
+        """
+        Return an `astropy.table.Table` representation of the query response.
+        """
+        return self
+
+    @property
+    @deprecated("2.1", "Slice the table instead.")
+    def blocks(self):
+        """
+        A `collections.abc.Sequence` object which contains the records
+        contained within the Query Response.
+        """
+        return list(self.iterrows())
 
+    @deprecated("2.1", "use path_format_keys instead")
     def response_block_properties(self):
-        client_name = self._client.__class__.__name__
-        warn(f"The {client_name} does not support response block properties.",
-             SunpyUserWarning)
-        return set()
+        return self.path_format_keys()
 
-    def append(self, table):
-        if self.table is None:
-            self.table = table
-        else:
-            self.table = vstack([self.table, table])
+    def unhide_columns(self):
+        """
+        Modify this table so that all columns are displayed.
+        """
+        self.display_keys = slice(None)
+        self.hide_keys = None
+        return self
+
+    @property
+    def _display_table(self):
+        """
+        Apply the display_keys and hide_keys attributes to the table.
+
+        This removes any keys in hide keys and then slices by any keys in
+        display_keys to return the correct table.
+        """
+
+        keys = list(self.colnames)
+        if self.hide_keys:
+            # Index only the keys not in hide keys in order
+            [keys.remove(key) for key in self.hide_keys if key in keys]
+
+        if self.display_keys != slice(None):
+            keys = [dk for dk in self.display_keys if dk in keys]
+
+        table = self[keys]
+        # The slicing operation resets display and hide keys to default, but we
+        # have already applied it
+        table.unhide_columns()
+
+        return table
+
+    def __str__(self):
+        """Print out human-readable summary of records retrieved"""
+        return '\n'.join(self._display_table.pformat(show_dtype=False))
+
+    def __repr__(self):
+        """Print out human-readable summary of records retrieved"""
+        return object.__repr__(self) + "\n" + str(self._display_table)
+
+    def _repr_html_(self):
+        return QTable._repr_html_(self._display_table)
 
     def show(self, *cols):
+        """
+        Return a table with only ``cols`` present.
```

Comment:
maybe this docstring should also say that it shows all available columns or something (i.e. `res[0].show() shows more colnames than just like `res[0]`, unless I'm missing something

Reply:
Yeah, some subclasses of this hide columns, i.e. fileid for VSO. I will add a sentence.